apps/web/src/components/VerificationCode/EnableNotificationDialog.tsx

```diff
@@ -0,0 +1,123 @@
+import { ChoosePhoneNumber } from "@/components/VerificationCode/EnterPhoneNumber";
+import { Typography } from "@litespace/ui/Typography";
+
+type EnableNotificationDialogProps = {
+  close: Void;
+  method: IUser.NotificationMethodLiteral | null;
+  phone: string | null;
+  sendCode: (payload: IConfirmationCode.SendVerifyPhoneCodePayload) => void;
+  sending: boolean;
+  verifyCode: (payload: IConfirmationCode.SendVerifyPhoneCodePayload) => void;
+  verifing: boolean;
+};
+
+type Step = "method" | "phone" | "code";
+
+export function EnableNotificationDialog({
+  close,
+  method,
+  phone,
+  sending,
+  sendCode,
+  verifyCode,
+  verifing,
+}: EnableNotificationDialogProps) {
+  const intl = useFormatMessage();
+  const [step, setStep] = useState<Step>();
+  const [activeMethod, setActiveMethod] =
+    useState<IUser.NotificationMethodLiteral | null>(method);
+  const [phoneNumber, setPhoneNumber] = useState<string | null>(phone);
+
+  const changeMethod = useCallback(
+    (method: IUser.NotificationMethodLiteral) => {
+      setActiveMethod(method);
+    },
+    []
+  );
+
+  const changeStep = useCallback((step: Step) => {
+    setStep(step);
+  }, []);
+
+  const sendVerificationCode = useCallback(() => {
+    if (!activeMethod || !phoneNumber) return;
+
+    sendCode({
+      method: activeMethod,
+      phone: phoneNumber,
+    });
+    changeStep("code");
+  }, [phoneNumber, activeMethod, changeStep, sendCode]);
+
+  // To auto detect the step
+  useEffect(() => {
+    if (step !== undefined) return;
+
+    if (phone && method) {
+      sendVerificationCode();
+      return;
+    }
+
+    if (method && !phone) {
+      setStep("phone");
+      return;
+    }
+
+    setStep("method");
+  }, [method, phone, sendVerificationCode, step]);
+
+  return (
+    <Dialog
+      title={
+        <Typography
+          tag="h2"
+          className="text-subtitle-1 font-bold text-natural-950"
+        >
+          {intl("notification-method.dialog.title")}
+        </Typography>
+      }
+      className="w-[512px] font-cairo"
+      open
+      close={close}
+    >
+      <AnimatePresence mode="wait">
+        {step === "method" ? (
+          <SelectMethod
+            activeMethod={activeMethod}
+            changeMethod={changeMethod}
+            changeStep={changeStep}
+            close={close}
+          />
+        ) : null}
+
+        {step === "phone" ? (
+          <ChoosePhoneNumber
+            close={close}
+            phoneNumber={phoneNumber}
+            sendVerificationCode={sendVerificationCode}
+            sending={sending}
+            setPhoneNumber={setPhoneNumber}
+          />
+        ) : null}
+
+        {step === "code" ? (
+          <VerifyCode
+            activeMethod={activeMethod}
+            phoneNumber={phoneNumber}
+            sendVerificationCode={sendVerificationCode}
+            verifyCode={verifyCode}
+            verifing={verifing}
+            close={close}
```

Comment:
The verify code component doesn't need access to the `method`.

```suggestion
            phoneNumber={phoneNumber}
            reseed={sendVerificationCode}
            verify={(code) => verifyCode({method: activeMethod, code})}
            verifing={verifing}
            close={close}
```